codeflash/code_utils/line_profile_utils.py

```diff
@@ -0,0 +1,237 @@
+    # todo modify by using a libcst transformer
+    module = cst.parse_module(original_code)
+    transformer = ProfileEnableTransformer(line_profile_output_file)
+    modified_module = module.visit(transformer)
+    return modified_module.code
+
+
+class ImportAdder(cst.CSTTransformer):
+    def __init__(self, import_statement):
+        self.import_statement = import_statement
+        self.has_import = False
+
+    def leave_Module(self, original_node, updated_node):
+        # If the import is already there, don't add it again
+        if self.has_import:
+            return updated_node
+
+        # Parse the import statement into a CST node
+        import_node = cst.parse_statement(self.import_statement)
+
+        # Add the import to the module's body
+        return updated_node.with_changes(
+            body=[import_node] + list(updated_node.body)
+        )
+
+    def visit_ImportFrom(self, node):
+        # Check if the profile is already imported from line_profiler
+        if node.module and node.module.value == "line_profiler":
+            for import_alias in node.names:
+                if import_alias.name.value == "profile":
+                    self.has_import = True
+
+
+def add_decorator_imports(function_to_optimize, code_context):
+    """Adds a profile decorator to a function in a Python file and all its helper functions."""
+    #self.function_to_optimize, file_path_to_helper_classes, self.test_cfg.tests_root
+    #grouped iteration, file to fns to optimize, from line_profiler import profile as codeflash_line_profile
+    file_paths = defaultdict(list)
+    line_profile_output_file = get_run_tmp_file(Path("baseline_lprof"))
+    file_paths[function_to_optimize.file_path].append(function_to_optimize.qualified_name)
+    for elem in code_context.helper_functions:
+        file_paths[elem.file_path].append(elem.qualified_name)
+    for file_path,fns_present in file_paths.items():
+        #open file
+        with open(file_path, "r", encoding="utf-8") as file:
+            file_contents = file.read()
+        # parse to cst
+        module_node = cst.parse_module(file_contents)
+        for fn_name in fns_present:
+            # add decorator
+            module_node = add_decorator_to_qualified_function(module_node, fn_name, 'codeflash_line_profile')
+        # add imports
+        # Create a transformer to add the import
+        transformer = ImportAdder("from line_profiler import profile as codeflash_line_profile")
+        # Apply the transformer to add the import
+        module_node = module_node.visit(transformer)
+        modified_code = isort.code(module_node.code, float_to_top=True)
+        # write to file
+        with open(file_path, "w", encoding="utf-8") as file:
+            file.write(modified_code)
+    #Adding profile.enable line for changing the savepath of the data, do this only for the main file and not the helper files
+    with open(function_to_optimize.file_path,'r') as f:
```

Comment:
dont use explicit open since it does not assume utf-8 encoding. We use pathlib paths in codeflash. so use something like Path(file_path).open():